From e33744e8cc82390153c94ace53c16f72365b9fd9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pavel=20B=C5=99ezina?= <pbrezina@redhat.com>
Date: Tue, 17 Jan 2017 11:58:06 +0100
Subject: [PATCH 23/79] ssh: do not create again fq name

We store fully qualified name in sysdb so there is no need to append
the domain part again which result in name@domain@domain string.
This field is not actually used in ssh client so it doesn't cause
any issue but we should stay correct here.

Reviewed-by: Jakub Hrozek <jhrozek@redhat.com>
---
 src/responder/ssh/sshsrv_cmd.c | 20 ++++++--------------
 1 file changed, 6 insertions(+), 14 deletions(-)

diff --git a/src/responder/ssh/sshsrv_cmd.c b/src/responder/ssh/sshsrv_cmd.c
index bd6270d0f1b62323ef7d140193351fb8585ce2ec..195d5763e9c5f4f9ff2f2f5ac49cd856d9198e7a 100644
--- a/src/responder/ssh/sshsrv_cmd.c
+++ b/src/responder/ssh/sshsrv_cmd.c
@@ -982,8 +982,7 @@ ssh_cmd_build_reply(struct ssh_cmd_ctx *cmd_ctx)
     struct ldb_message_element *el_user_cert_keys = NULL;
     uint32_t count = 0;
     const char *name;
-    char *fqname;
-    uint32_t fqname_len;
+    uint32_t name_len;
     TALLOC_CTX *tmp_ctx;
     struct ssh_ctx *ssh_ctx;
     struct cli_protocol *pctx;
@@ -1060,38 +1059,31 @@ ssh_cmd_build_reply(struct ssh_cmd_ctx *cmd_ctx)
         goto done;
     }
 
-    fqname = talloc_asprintf(cmd_ctx, "%s@%s",
-                             name, cmd_ctx->domain->name);
-    if (!fqname) {
-        ret = ENOMEM;
-        goto done;
-    }
-
-    fqname_len = strlen(fqname)+1;
+    name_len = strlen(name) + 1;
 
     ret = decode_and_add_base64_data(cmd_ctx, el, false, ssh_ctx,
-                                     fqname_len, fqname, &c);
+                                     name_len, name, &c);
     if (ret != EOK) {
         DEBUG(SSSDBG_OP_FAILURE, "decode_and_add_base64_data failed.\n");
         goto done;
     }
 
     ret = decode_and_add_base64_data(cmd_ctx, el_orig, false, ssh_ctx,
-                                     fqname_len, fqname, &c);
+                                     name_len, name, &c);
     if (ret != EOK) {
         DEBUG(SSSDBG_OP_FAILURE, "decode_and_add_base64_data failed.\n");
         goto done;
     }
 
     ret = decode_and_add_base64_data(cmd_ctx, el_override, false, ssh_ctx,
-                                     fqname_len, fqname, &c);
+                                     name_len, name, &c);
     if (ret != EOK) {
         DEBUG(SSSDBG_OP_FAILURE, "decode_and_add_base64_data failed.\n");
         goto done;
     }
 
     ret = decode_and_add_base64_data(cmd_ctx, el_user_cert_keys, true, ssh_ctx,
-                                     fqname_len, fqname, &c);
+                                     name_len, name, &c);
     if (ret != EOK) {
         DEBUG(SSSDBG_OP_FAILURE, "decode_and_add_base64_data failed.\n");
         goto done;
-- 
2.9.3

